matrix/plugins/actions.py

```diff
@@ -0,0 +1,45 @@
+from datetime import timedelta, datetime
+
+from .glitch import glitch  # noqa
+
+
+async def deploy(context, rule, action, event=None):
+    rule.log.info("Deploying %s", action.args['entity'])
+    context.apps.extend(await context.juju_model.deploy(action.args['entity']))
+    rule.log.info("Deploy COMPLETE")
+    return True
+
+
+async def health(context, rule, action, event=None):
+    if not context.apps:
+        # this shouldn't happen, because the health rule has
+        # an "after: deploy", but that doesn't seem to be honored
+        return
+    stable_period = timedelta(seconds=30)
+    errored_units = []
+    busy_units = []
+    for app in context.apps:
+        for unit in app.units:
+            now = datetime.utcnow()
+            agent_status_duration = now - unit.agent_status_since
```

Comment:
We certainly won't remain in a stable state, but my intention was to gate glitch on seeing `deployment.healthy` at least once before starting (doesn't currently work, due to `after:` not working).  Possibly we should not switch to `deployment.busy` if it was already at `deployment.healthy` and only allow a transition from that to `deployment.unhealthy` instead.